services/registration/src/models/application.ts

```diff
@@ -7,7 +7,31 @@ export interface Application {
   userId: string;
   hexathon: Types.ObjectId;
   applicationBranch: Types.ObjectId;
-  applicationData: Mixed;
+  applicationData: {
+    adult: boolean;
+    occupation: string;
+    school: string;
+    graduationYear: number;
+    major: string;
+    shirtSize: string;
+    dietaryRestrictions: string;
+    phoneNumber: string;
+    gender: string;
+    ethnicity: string;
+    address: {
+      line1: string;
+      line2: string;
+      city: string;
+      state: string;
+      zip: number;
+      country: string;
+    };
+    website: string;
+    linkedin: string;
+    extraInfo: [Types.ObjectId];
+    confirmChecks: Types.ObjectId;
+    essays: [Types.ObjectId];
```

Comment:
So this stuff shouldn't be `objectId`. Object Id only refers to a string. Referencing the issue, this stuff should be objects, so you should use Types.Mixed and reference the Mongoose documentation as well